assets/admin/exit-survey/form.js

```diff
@@ -0,0 +1,67 @@
+import { useCallback, useRef, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { ExitSurveyFormItem } from './form-item';
+import { reasons } from './reasons';
+
+/**
+ * Exit survey form.
+ *
+ * @param {Object}   props
+ * @param {Function} props.submit Callback to submit feedback.
+ * @param {Function} props.skip   Callback to skip feedback.
+ */
+export const ExitSurveyForm = ( { submit, skip } ) => {
+	const form = useRef( null );
+	const [ , updateInput ] = useState( null );
+
+	const submitForm = useCallback(
+		( e ) => {
+			e.preventDefault();
+			const formData = new window.FormData( form.current );
+			const reason = formData.get( 'reason' );
+			submit( {
+				reason,
+				details: reason && formData.get( `details-${ reason }` ),
+			} );
+		},
+		[ submit ]
+	);
+
+	const hasInput = !! form.current?.elements?.reason?.value;
+
+	return (
+		<form
+			onChange={ updateInput }
+			className="sensei-modal sensei-exit-survey"
+			ref={ form }
+			onSubmit={ submitForm }
+		>
+			<div className="sensei-exit-survey__content">
+				<h2>{ __( 'Quick Feedback', 'sensei-lms' ) }</h2>
+				<p>
+					{ __(
+						'If you have a moment, please let us know why you are deactivating so that we can work to improve our product.',
+						'sensei-lms'
+					) }
+				</p>
+				{ reasons.map( ExitSurveyFormItem ) }
```

Comment:
Interesting! I have never seen this approach to map directly in the component instead of the instance. I didn't know it worked!
But while debugging, I noticed that React dev tools don't recognize the mapped components. Do you know a way to fix that? Or maybe we should map it using the component instance?

Reply:
Yeah, this is just a function, so it works. But updated it to component instance to make it clearer.

Reply:
I think with this change we can also remove the `key={ label }` from the `ExitSurveyFormItem` render, right?

Reply:
Yup, updated in 180ed8b